plumpy/processes.py

```diff
@@ -738,7 +738,9 @@ def on_finished(self):
 
     @super_check
     def on_except(self, exc_info):
-        self.future().set_exception(exc_info[1])
+        exception = exc_info[1]
+        exception.__traceback__ = exc_info[2]
+        self.future().set_exception(exception)
```

Comment:
Just a question: is this really the recommended way of doing this (i.e. peeling out things from `exc_info`?)
Is there no wrapper that transforms `exc_info` into something you can pass to `future().set_exception`?

Reply:
I am only doing this because the future here is a tornado `concurrent.Future` and the implementation of `set_exception` is as follows:
```

    def set_exception(self, exception):
        """Sets the exception of a ``Future.``"""
        self.set_exc_info(
            (exception.__class__,
             exception,
             getattr(exception, '__traceback__', None)))
```

Reply:
I see that the `exc_info` is something homegrown:
https://github.com/aiidateam/plumpy/blob/9a6ab9f0c4e1b315cdf1a3cc0b3ddaafb6248c30/plumpy/process_states.py#L362

Should we open an issue to consider using a standard exception class rather than separate values for exception type, message and traceback?

Anyhow, this explains why you need to do manual work here...

Reply:
No that is actually standard python. If you call `sys.exc_info` it will return a tuple of exception type, exception and traceback. This is just a utility function to returns the stored version, because the system can override the current one if another exception occurs.